### ISSUE TITLE
Malformed YAML with autoscalingGroupTags

### ISSUE THREAD

**Issue by lukehoban** [pulumi/pulumi-eks#146]
From https://pulumi-community.slack.com/archives/C84L4E3N1/p1559680345069300:

here's what i've got

```
autoScalingGroupTags: {
                [`kubernetes.io/cluster/${clusterName}`]: "",
                "k8s.io/cluster-autoscaler/enabled": "",
}
```

Results in:

```
Plan apply failed: Creating CloudFormation stack failed: ValidationError: Template format error: YAML not well-formed. (line 51, column 1)
```

**Comment by metral** [comment]
> but I still hit a yaml templating error for [`kubernetes.io/cluster/${clusterName}`]

I'm not quite understanding the issue here - it appears you're trying to use brackets `[]` as part of the tag key, but that char is not one of the ones allowed: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-resource-tags.html.

Have you tried passing in `kubernetes.io/cluster/${clusterName}` as a fully-rendered string?

**Comment by HansMoleman** [comment]
The tag that is added by pulumi/eks is different than the required tag for cluster autoscaling. pulumi/eks will add the  `kubernetes.io/cluster/${clusterName}` tag but for auto scaling we also need the  `k8s.io/cluster-autoscaler/${clusterName}` tag. There's a slight difference in the provided tag and the required tag.

Since we are needing to dynamically generate the key for this tag, typescript wants to wrap that in `[]`. I tried to save the fully rendered string as a variable to then use a key; however, variables are not valid keys.

wrapping the key in `[]` should just render as a string. Consider the following code sample


```const myAttributeName = 'foo';
// Does not create an object that looks like {foo: 'somValue'}
const somObj = {
    myAttributeName: 'someValue',
};
// Does create {foo: 'someValue'}
const otherObj = {
    [myAttributeName]: 'someValue',
}```

**Comment by HansMoleman** [comment]
Just realized that there's a typo in the tag that I'm wanting to apply. I'll edit the comments so that they are correct

**Comment by HansMoleman** [comment]
I've updated to `@pulumi/eks@0.18.6` and I'm still hitting this error. Can we reopen this bug?

Here's the code I'm trying to run:
```
const nodeGroup = cluster.createNodeGroup(`${config.environment}-default-worker-node-group-${i}`, {
            instanceType: config.instanceType,
            nodeSubnetIds: [subnetIds.ids[i]],
            nodePublicKey: config.workerNodePublicKey,
            nodeRootVolumeSize: config.ebsRootVolumeSize,
            desiredCapacity: config.desiredNodes,
            maxSize: config.maxNumberOfNodes,
            minSize: config.minNumberOfNodes,
            nodeUserData: userData,
            amiId: config.amiId,
            instanceProfile: defaultInstanceProfile,
            autoScalingGroupTags: {
                [`kubernetes.io/cluster/${clusterName}`]: "true",
                "k8s.io/cluster-autoscaler/enabled": "true",
            }
        });
```

Everything works fine except for:
```
[`kubernetes.io/cluster/${clusterName}`]: "true"
```

Here's the error:
```
Diagnostics:
  aws:cloudformation:Stack (online-sandbox-bngo-default-worker-node-group-1-nodes):
    error: Plan apply failed: Creating CloudFormation stack failed: ValidationError: Template format error: YAML not well-formed. (line 51, column 1)
    	status code: 400, request id: 87286cd5-8d59-11e9-8304-55fd988798ff

  pulumi:pulumi:Stack (eks-cluster-online-sandbox-bngo):
    error: update failed

  aws:cloudformation:Stack (online-sandbox-bngo-default-worker-node-group-0-nodes):
    error: Plan apply failed: Creating CloudFormation stack failed: ValidationError: Template format error: YAML not well-formed. (line 51, column 1)
    	status code: 400, request id: 8728bbc8-8d59-11e9-81c3-d14669dcbb2b
```

**Comment by metral** [comment]
The [fix](https://github.com/pulumi/pulumi-eks/pull/157) for this bug just landed into master < 1 hour ago and is still waiting for CI to clear, before a new release can be cut that includes it. As soon as a new release is done, I'll be sure to link it here.

**Comment by HansMoleman** [comment]
Perfect, thanks!

**Comment by metral** [comment]
v0.18.7 has been [released](https://github.com/pulumi/pulumi-eks/blob/master/CHANGELOG.md#0187-released-june-12-2019), which includes the [fix](#157 ) for this issue.
